core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -597,7 +598,10 @@ private ClientFactoryOptions buildOptions() {
                         if (dnsResolverGroupCustomizers != null) {
                             dnsResolverGroupCustomizers.forEach(consumer -> consumer.accept(builder));
                         }
-                        return builder.build(eventLoopGroup);
+                        return builder
+                                .meterRegistry(options.containsKey(ClientFactoryOptions.METER_REGISTRY) ?
+                                        (MeterRegistry) options.get(ClientFactoryOptions.METER_REGISTRY).value()
+                                        : PrometheusMeterRegistries.newRegistry()).build(eventLoopGroup);
```

Comment:
Could avoid using `PrometheusMeterRegistries` here, because we don't have a hard dependency on it. It has to be like the following IIUC:

```java
return builder.meterRegistry(options.meterRegistry()).build(eventLoopGroup);
```

Reply:
options is an Map<ClientFactoryOption<?>, ClientFactoryOptionValue<?>> options in this class.   Also the reason I was doing the null check is because there is a static ClientFactory that is being instantiated inside the https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/client/DefaultClientFactory.java#L72 and that does not have the options Map built up.

Reply:
Ahh, OK. Then could we use `ClientFactoryOptions.of().meterRegistry()` instead of `PrometheusMeterRegistries.newRegistry()`, so that the default registry is chosen automatically?